src/server/stream_family.cc

```diff
@@ -3130,6 +3135,14 @@ variant<bool, facade::ErrorReply> HasEntries2(const OpArgs& op_args, string_view
       requested_sitem.id.val = requested_sitem.group->last_id;
       streamIncrID(&requested_sitem.id.val);
     }
+  } else {
+    // Resolve $ to the last ID in the stream.
+    if (requested_sitem.id.resolve_last_id) {
+      requested_sitem.id.val = last_id;
+      streamIncrID(&requested_sitem.id.val);  // include id's strictly greater
```

Comment:
should we check the result of streamIncrID().

Reply:
I backported the changes from valkey. They do not check it.

Reply:
we check it sometimes and sometimes no. We can rewrite this method in this case and never check for example